server/database.go

```diff
@@ -153,6 +153,11 @@ func (s *Server) registerDatabaseRoutes(g *echo.Group) {
 		if database == nil {
 			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Database not found with ID %d", id))
 		}
+		// Wildcard(*) database is used to connect all database at instance level.
+		// Do not return it via `get database by id` API.
+		if database.Name == api.AllDatabaseName {
+			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Database not found with ID %d", id))
```

Comment:
If the client explicitly requests this all database, then we should return it.

I am wondering in which case the client needs to request this "all databases"?

Reply:
Check out this comment: https://github.com/bytebase/bytebase/pull/881#pullrequestreview-910371797
It may be a mistake that when we patched/created a data source, then request its related * database.

Reply:
yeah, I guess we do not need to request the related * database, instead we request the whole instance which will grab the data source?

IIUC, the problem is if we request the database, then that database will be stored in the frontend cached map, then it will show up in the instance's database list.

Reply:
Yes, but I do the changes in this PR that if client get * database by id, server will return not found. So there is no possible to get a * database in API level.